### DISCUSSION TITLE
Alerts - Does this plugin support alerts?

### DISCUSSION THREAD

**Discussion by GiovanniSinosini** [grafana/grafana-infinity-datasource#189] [Q&A] [answered]
Does this plugin support alerts?

Idea => Under "Thresholds Options", there could be a pop-up (alert) creator, for example:

If the variable "xpto" is different from 0, start the pop-up (string alert). 

![image](https://user-images.githubusercontent.com/56783045/139541212-49b5b85c-58b1-41d8-9a76-efc412de41ec.png)

**Comment by yesoreyeram** [comment]
No .it doesn't support alerts unfortunately. Due to the dynamic handling of data in JavaScript, most of the logic happens in frontend. Only the core data retrieval happens in backend due to security.  In order to support alerts data parsing needs to be in backend. 

This is one of the compromise we did to support multiple data types. But I don't understand what you mean by string alert.

**Reply by GiovanniSinosini** [reply]
String alert => When the condition in the "threshold option" is satisfied, a new window (or pop-up) opens with a message that has been edited by the user.

With the "threshold option" we can change the color of the variable or the panel.

Then, with the same logic of changing colors, we could start a JavaScript code in the frontend to open a new window (or popup) with the message edited by the user without having to analyze the backend.

Below is a link to an example popup:
https://www.quackit.com/html/codes/html_popup_window_code.cfm  

Idea:
If the variable "xpto" is different from 0, start the JavaScript code (popup).

I think it can be the solution for alerts with edited messages from users without major plugin changes.

**Comment by GiovanniSinosini** [comment]
![missaoQUASEcumprida](https://user-images.githubusercontent.com/56783045/151591124-c41aa411-26db-4ed3-a3c9-0690db737774.gif)


Alert with popup created successfully.   :  )

**Reply by yesoreyeram** [reply]
nice. may be adding more detail about how you did this may help others.

**Comment by GiovanniSinosini** [comment]
Of course yes. I implemented the above idea (on October 30, 2021).
I created an external plugin to capture the screen data with javascript and used the browser's localStorage to store the data.
But the main idea of ​​the solution was described above (on Oct 30, 2021).

**Comment by yesoreyeram** [comment, marked as answer]
1.0.0-beta.1 is now **[available](https://github.com/yesoreyeram/grafana-infinity-datasource/releases/tag/v1.0.0-beta.1)** with alerting support. 

With JSON query type, one need to choose `backend` as a parser to get the alerts working. Example shown below for setting up alert.

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/153843/188924894-4b1f3d3d-2857-484e-916c-d09cb8d4dd5c.png">

Example API used : https://thingspeak.com/channels/38629/feed.json
